codesrc/src/main/java/com/onehealth/processors/AddRunsRequestProcessor.java

```diff
@@ -30,11 +32,15 @@
 	@Autowired
 	KafkaUtils kafkaUtils;
 
+	private static final Logger LOG = Logger.getLogger(AddRunsRequestProcessor.class);
+
 	@Override
-	public boolean isRequestValid(AddRunDetailsRequest request) throws Exception {
+	public boolean isRequestValid(AddRunDetailsRequest request) {
 		if (StringUtils.isEmpty(request.getUserId())) {
-			throw new Exception("User Id is not populated");
+			LOG.error("Invalid user id found, unable to process request");
+			throw new NoUserFoundException();
 		}
+		LOG.debug("Valid add run details request, processing..");
 		return super.isRequestValid(request);
```

Comment:
Also, please add logging to rest controller, just print the endpoint path and queryparams and the type of endpoint (GET/POST)